benchmark/rocblas/blas3/symm.cpp

```diff
@@ -0,0 +1,206 @@
+      << side << "/" << uplo << "/" << m << "/" << n << "/" << alpha << "/"
+      << beta;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void rocblas_symm_f(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CHECK_ROCBLAS_STATUS(rocblas_ssymm(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CHECK_ROCBLAS_STATUS(rocblas_dsymm(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, rocblas_handle& rb_handle, char side,
+         char uplo, index_t m, index_t n, scalar_t alpha, scalar_t beta,
+         bool* success) {
+  // Standard test setup.
+  index_t lda = (side == 'l') ? m : n;
+  index_t ldb = m;
+  index_t ldc = ldb;
+
+  {
+    // The counters are double. We convert m, n and k to double to avoid
+    // integer overflows for n_fl_ops and bytes_processed
+    double m_d = static_cast<double>(m);
+    double n_d = static_cast<double>(n);
+
+    state.counters["m"] = m_d;
+    state.counters["n"] = n_d;
+
+    const double mem_readB = m_d * n_d;
+    const double mem_readC = (beta != scalar_t{0}) ? m_d * n_d : 0;
+    const double mem_writeC = m_d * n_d;
+    const double mem_readA =
+        (side == 'l') ? (m_d * (m_d + 1) / 2) : (n_d * (n_d + 1) / 2);
+    const double total_mem =
+        (mem_readB + mem_readC + mem_writeC + mem_readA) * sizeof(scalar_t);
+    state.counters["bytes_processed"] = total_mem;
+
+    const double nflops_AtimesB =
+        (side == 'l') ? (2 * m_d * m_d) * n_d : 2 * n_d * n_d * n_d;
+    const double nflops_addBetaC = 2 * m_d * n_d;
+    const double total_nflops = nflops_AtimesB + nflops_addBetaC;
+    state.counters["n_fl_ops"] = total_nflops;
+
+    state.SetBytesProcessed(state.iterations() * total_mem);
+    state.SetItemsProcessed(state.iterations() * total_nflops);
+  }
+
+  // Matrix options (rocBLAS)
+  const rocblas_side side_rb =
+      (side == 'l') ? rocblas_side_left : rocblas_side_right;
+  const rocblas_fill uplo_rb =
+      (uplo == 'u') ? rocblas_fill_upper : rocblas_fill_lower;
+
+  // Data sizes
+  const int a_size = lda * lda;
+  const int b_size = ldb * n;
+  const int c_size = ldc * n;
+
+  // Matrices
+  std::vector<scalar_t> a =
+      blas_benchmark::utils::random_data<scalar_t>(a_size);
+  std::vector<scalar_t> b =
+      blas_benchmark::utils::random_data<scalar_t>(b_size);
+  std::vector<scalar_t> c =
+      blas_benchmark::utils::random_data<scalar_t>(c_size);
+
+  {
+    // Device memory allocation & H2D copy
+    blas_benchmark::utils::HIPVector<scalar_t> a_gpu(a_size, a.data());
+    blas_benchmark::utils::HIPVector<scalar_t> b_gpu(b_size, b.data());
+    blas_benchmark::utils::HIPVector<scalar_t> c_gpu(c_size, c.data());
+
+    CHECK_ROCBLAS_STATUS(
+        rocblas_set_pointer_mode(rb_handle, rocblas_pointer_mode_host));
```

Comment:
Thank you for your work @OuadiElfarouki ! 
Do we need to set the pointer-mode for  this operator?